tezos_client/src/client.rs

```diff
@@ -0,0 +1,107 @@
+use failure::Fail;
+
+use networking::p2p::binary_message::Hexable;
+use networking::p2p::encoding::prelude::*;
+use tezos_encoding::hash::{BlockHash, ChainId};
+use tezos_interop::ffi;
+use tezos_interop::ffi::OcamlStorageInitInfo;
+
+#[derive(Debug, Fail)]
+pub enum ApplyBlockError {
+    #[fail(display = "incomplete operations")]
+    IncompleteOperations,
+    #[fail(display = "failed to apply block")]
+    FailedToApplyBlock,
+    #[fail(display = "Unknown predecessor")]
+    UnknownPredecessor,
+}
+
+pub struct TezosStorageInitInfo {
+    pub chain_id: ChainId,
+    pub genesis_block_header_hash: BlockHash,
+    pub current_block_header_hash: BlockHash
+}
+
+impl TezosStorageInitInfo {
+    fn new(storage_init_info: OcamlStorageInitInfo) -> Self {
+        TezosStorageInitInfo {
+            chain_id: hex::decode(storage_init_info.chain_id).unwrap(),
+            genesis_block_header_hash: hex::decode(storage_init_info.genesis_block_header_hash).unwrap(),
```

Comment:
We should now call `.unwrap()` on results at this level because this will make the whole application to panic. Instead return `Result<..>`.